### PR TITLE
Create Dockerfile

### PR DESCRIPTION
**Pull request by rafinskipg**
Hello 

I want to add an example Dockerfile that works with this project, so people can deploy under docker enviroments. 

I got a problem with the `NODE_ENV=production` since the package `cross-env` is under `devDependencies` and the build stage needs it.  So i had to run the command `yarn install` witha  NODE_ENV set to `development`, anyway. This should work for deploying this project somewhere.

I can add some explanation in the README on how to run it if you accept this on the project

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Bundle app source
+COPY . /usr/src/app
+
+RUN NODE_ENV=development yarn install && yarn cache clean
+
+RUN NODE_ENV=production yarn build
```
Comment (eddyerburgh) [inline Dockerfile]: Since NODE_ENV is set to production in line 4, do you need to set it again here?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FROM node:8.12
```
Comment (eddyerburgh) [inline Dockerfile]: Can you make this node 10, so that it stays up-to-date for longer

**Comment by eddyerburgh** [comment]
Thanks for this, I can see it being useful for others 👍 

I've left a couple of comments

File: Dockerfile (hunk 1 above)
Comment (rafinskipg) [reply Dockerfile]: I think that the line 12 NODE_ENV=development, makes it change, but let me confirm

File: Dockerfile (hunk 2 above)
Comment (rafinskipg) [reply Dockerfile]: Sure

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ENV PORT 8080
+ENV NODE_ENV production
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Bundle app source
+COPY . /usr/src/app
+
+# We need to use NODE_ENV development because we need the dependency `cross_env` for the build and it is listed under devDependencies
```
Comment (eddyerburgh) [inline Dockerfile]:
```suggestion
# Install all dependencies, including devDependencies for the build job
```

**Pull request merged**